pkg/util/history/controller_history_test.go

```diff
@@ -69,7 +71,13 @@ func TestRevisionHistory(t *testing.T) {
 		t.Fatalf("Failed to new controller revision: %v", err)
 	}
 
-	fakeClient := fake.NewClientBuilder().Build()
+	fakeClient := fake.NewClientBuilder().WithIndex(&apps.ControllerRevision{}, fieldindex.IndexNameForOwnerRefUID, func(obj client.Object) []string {
+		var owners []string
```

Comment:
There are so many errors in other uts and can be fixed by using client with index.
I think we should wrap this code with a customized function and replace `fake.NewClientBuilder()`.